File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/service/value/factory/AttValueFactory.java
package ru.citeck.ecos.records3.record.op.atts.service.value.factory;

import ru.citeck.ecos.records3.record.op.atts.service.value.AttValue;

import java.util.List;

public interface AttValueFactory<T> {

    AttValue getValue(T value);

    List<Class<? extends T>> getValueTypes();
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/service/value/factory/AttValueFactory.kt
package ru.citeck.ecos.records3.record.op.atts.service.value.factory

import ru.citeck.ecos.records3.record.op.atts.service.value.AttValue

interface AttValueFactory<T> {

    fun getValue(value: T): AttValue?

    fun getValueTypes(): List<Class<*>>
}
